### ISSUE TITLE
Support building with Rust stable

### ISSUE THREAD

**Issue by orhun** [Cretezy/lazyjj#41]
Hello! 👋🏼 

I recently packaged `lazyjj` for Arch Linux (#40) - I had to use Rust nightly because of this feature:

```
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> src/main.rs:1:1
  |
1 | #![feature(let_chains)]
  | ^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
error: could not compile `lazyjj` (bin "lazyjj") due to 1 previous error
```

On Arch we prefer to use the stable channel because it helps with [reproducibility](https://reproducible-builds.org/) of the packages.

Would you be interested in supporting Rust stable by removing that feature and refactoring the code? 🐻

**Comment by Cretezy** [comment]
I would like to support stable, however the codebase currently uses let chains a lot (there are 31 usages currently).

I think for the moment, until lazyjj stabilizes, it will stay on nightly. I'll look into migrating the let chains when I have extra time.

Thank you!

**Comment by orhun** [comment]
Sure thing, that sounds good.

I'm also interested in contributing, so let me know when you are interested in a PR 🐻

**Comment by Cretezy** [comment]
I think we can go ahead and migrate the Rust stable. This would make packaging a lot simpler.

**Comment by maralorn** [comment]
At least for nixos I have been informed there is probably a viable solution to build the package as is. Will report back.

**Comment by Cretezy** [comment]
Fixed in [v0.3.1](https://github.com/Cretezy/lazyjj/releases/tag/v0.3.1).

**Comment by 0-wiz-0** [comment]
Thank you!
